### PR TITLE
Deprecate quarkus-bootstrap-maven-plugin in favor of quarkus-extension-maven-plugin

### PR DESCRIPTION
**Pull request by aloubyansky**
The `quarkus-bootstrap-maven-plugin` includes a single goal `extension-descriptor` which is used in extension projects to generate extension metadata, etc. I thought it would make more sense to have a more appropriate artifactId for the plugin.

This PR deprecates the `quarkus-bootstrap-maven-plugin` and updates everything (the extension POMs, docs, project templates and tests) to use `quarkus-extension-maven-plugin` instead.

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]

File: extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/runtime/log [hunk 1]
```diff
@@ -0,0 +1,109 @@
+[INFO] Scanning for projects...
```
Comment (gastaldi) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/runtime/log:1]: Is this file meant to be included here?

Comment (aloubyansky) [reply extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/runtime/log:1]: oh, good catch, no, i'll remove it

**Comment by rsvoboda** [comment]
@aloubyansky feels like noteworthy label candidate

**Comment by aloubyansky** [comment]
Thanks @rsvoboda

**Comment by rsvoboda** [comment]
there is one side-effect to this change, only people using quarkus cli from quarkus main would notice

```
qss -version                                                                                                                            ✔  11:10:03 
999-SNAPSHOT

qss create extension ext
cd quarkus-ext
mvn clean package

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Quarkus - Ext - Parent 1.0.0-SNAPSHOT:
[INFO]
[INFO] Quarkus - Ext - Parent ............................. SUCCESS [  2.060 s]
[INFO] Quarkus - Ext - Runtime ............................ FAILURE [  2.004 s]
[INFO] Quarkus - Ext - Deployment ......................... SKIPPED
[INFO] Quarkus - Ext - Integration Tests .................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.345 s
[INFO] Finished at: 2022-06-08T11:09:02+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Plugin io.quarkus:quarkus-extension-maven-plugin:2.9.2.Final or one of its dependencies could not be resolved: Could not find artifact io.quarkus:quarkus-extension-maven-plugin:jar:2.9.2.Final in jboss-public-repository (https://repository.jboss.org/nexus/content/groups/public/) -> [Help 1]

```

It's because the version of CLI defines what code-starters get applied and the registry defines what (released) version gets used.
So the 999-SNAPSHOT based CLI forces extension which is not available in 2.9.2.Final release.

**Pull request merged**